compiler/rustc_codegen_ssa/src/mir/debuginfo.rs

```diff
@@ -301,7 +293,40 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                 }
             }
 
-            bx.dbg_var_addr(dbg_var, dbg_loc, base.llval, direct_offset, &indirect_offsets);
+            // When targeting MSVC, spill extra locals for arguments instead of pointing multiple
+            // dbg_var_addr() calls into the same alloca with offsets. MSVC uses CodeView records
+            // not DWARF and LLVM doesn't support translating the resulting
+            // [DW_OP_deref, DW_OP_plus_uconst, offset, DW_OP_deref] debug info to CodeView.
+            // Spilling extra values to the stack makes the resulting debug info simple enough
```

Comment:
This seems somewhat inaccurate as AFAICT what we're doing is creating an alloca containing a pointer to a place we're debug-infoing rather than actually spilling the values themselves (which are very likely to already be "spilled" via an alloca in the first place).

In other words, it seems to suggest that the issue is in the fact that a value is not spilled, but in reality the issue is that setting debug info to most fields within a value doesn't really work, if I understand correctly, right?

Reply:
Yeah, that's correct. I'll update the comment with better terminology.